src/Compilers/Core/Portable/Diagnostic/LocalizableString.cs

```diff
@@ -32,22 +34,22 @@ public string ToString(IFormatProvider formatProvider)
             }
         }
 
-        public static explicit operator string(LocalizableString localizableResource)
+        public static explicit operator string?(LocalizableString localizableResource)
         {
             return localizableResource.ToString(null);
         }
 
-        public static implicit operator LocalizableString(string fixedResource)
+        public static implicit operator LocalizableString(string? fixedResource)
         {
             return FixedLocalizableString.Create(fixedResource);
         }
 
-        public sealed override string ToString()
+        public sealed override string? ToString()
```

Comment:
😭 

Consider making the `ToString` methods return an empty string instead of null.

Reply:
This behavior is too ingrained into the system. I agree it's not ideal though. If we wanted to file a follow up issue to remove this and clean up tests, behavior, etc ... I'd be supportive of that. But do want to try and keep annotations separate from large refactorings. Otherwise the annotation cost is going to sky rocket in places.

---
In reply to: [356231737](https://github.com/dotnet/roslyn/pull/40268#discussion_r356231737) [](ancestors = 356231737)